proposals/NNNN-Addition-of-a-way-to-notify-HMI-of-the-required-support-for-apps.md

```diff
@@ -0,0 +1,103 @@
+| :- | :- | :- |
+| 0x0a | Request Additional Support | Requests that the additional supports that SDL app needs. |
+| 0x0b | Ping Apps | Acknowledges that SDL app is available the additional supports. |
+
+<b>Table 2.</b> The parameters of `Request Additional Support` on Protocol spec
+| Tag Name | Type | Introduced | Description |
+| :- | :- | :- | :- |
+| appName | String | 5.5.0 | The name of SDL app that needs the additional supports. |
+| requiredAdditionalSupport | String Array | 5.5.0 | An array of the additional suuport names which SDL app needs.<br>- audioSupport<br>- highBandwidth |
+
+Note that the following additional support items are described. 
+ - audioSupport : To add `audioSupport` to the parameter of `requiredAdditionalSupport' if the system does not have the support to play audio when an app that needs to play audio, such as Media app, will be registered.
+ - highBandwidth : To add `highBandwidth` to the parameter of `requiredAdditionalSupport' if the system does not have the support the high band width when an app that needs the video projection, such as Navigation app, will be registered.
+It is expandable by adding the string corresponded the additional support that the SDL app needs.
+
+<b>Table 3.</b> The parameters of `Ping Apps` on Protocol spec
+
+| Tag Name | Type | Introduced | Description |
+| :- | :- | :- | :- |
+| appName | String | X.X.X.X | The name of SDL app that uses the additional supports. |
+
+#### HMI_API
+The HMI_API `requestAdditionalSupport` has two parameters `appName` and `requiredAdditionalSupport`. Also, the HMI_API `pingApps` has a parameters `appName`. The description of these parameter is shown as below.
+
+```xml
++   <function name="RequestAdditionalSupport" messagetype="notification">
++       <description>Notification from SDL to HMI to notify the additional supports that SDL app needs.</description>
++       <param name="appName" type="String" maxlength="100" mandatory="true">
++           <description>The name of SDL app that needs the additional supports.</description>
++       </param>
++       <param name="requiredAdditionalSupport" type="String" minsize="0" maxsize="100" maxlength="100" array="true" mandatory="true">
++           <description>An array of the additional suuport names which SDL app needs. Ex, audioSupport, highBandwidth.</description>
++       </param>
++   </function>
+
++   <function name="PingApps" messagetype="notification">
++       <description>Notification from HMI to SDL to notify that SDL app is available the additional supports.</description>
++       <param name="appName" type="String" maxlength="100" mandatory="true">
++           <description>The name of SDL app that uses the additional supports.</description>
++       </param>
++   </function>
+```
+
+#### Modification of Router Service
+To realize this proposal, it is necessary to modify the Router service as follows.
+ - Receive the `Ping Apps` message from SDL Core at Protocol Spec.
```

Comment:
Please reword. I'm unsure what is meant by "...from SDL Core at Protocol Spec."

Reply:
We will modify to "Receive the Protocol Spec message `Ping Apps`  from SDL Core".